packages/lib/onedrive-api.ts

```diff
@@ -265,11 +266,12 @@ export default class OneDriveApi {
 
 		for (let i = 0; i < 5; i++) {
 			options.headers['Authorization'] = `bearer ${this.token()}`;
+			options.headers['User-Agent'] = `ISV|Joplin|Joplin/${version}`;
```

Comment:
I'm wondering here, should we put the lib version number or the app version number? It seems the app version would be more accurate? It can be retrieved using shim.appVersion().

Also I don't know the format of this user agent string, but shouldn't it be different for mobile/cli/desktop? Such as Joplin-Desktop, Joplin-Android, Joplin-iOS, Joplin-Cli (not sure about the format)? The platform is needed is in shim.platformName()

Reply:
shim.appVersion() is great! I used the lib version as I didn't want to pull from another package and I honestly didn't look deep enough into shim to notice the functions being overloaded.

As for the user-agent, I'm just following the Microsoft Docs. The response returned is the same for all platforms so I don't think the user-agent being the same is an issue for OneDrive API requests. Unless you can think of a reason we should delineate between the apps?

**Source**
- https://docs.microsoft.com/en-us/sharepoint/dev/general-development/how-to-avoid-getting-throttled-or-blocked-in-sharepoint-online#how-to-decorate-your-http-traffic-to-avoid-throttling

![image](https://user-images.githubusercontent.com/32174181/127323629-12648796-a2b7-4a2e-96da-1cf3cb79e093.png)

The documentation is a bit strange, this article (above) is written for SharePoint but mentions OneDrive. Also, OneDrive is essentially a personal SharePoint site and operates under the [same api](https://docs.microsoft.com/en-us/onedrive/developer/rest-api/?view=odsp-graph-online). So we are trying to connect the dots and assume this change should help with the OneDrive throttling as well.

Reply:
Yes your user-agent is right then based on their documentation, it should be just "Joplin" as app name. And then let's used shim.appVersion for the version field then.